### PR TITLE
Add 'to do List' project

### PR DESCRIPTION
**Pull request by VeraPavlova18**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
+import React, { Component } from 'react'
+import { FlatList, Text, StyleSheet, View, Button, Alert } from 'react-native'
+import TextInputToDoList from './src/components/TextInputToDoList'
+import { v4 } from 'uuid'
+
+export default class App extends Component {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  constructor (props) {
+    super(props)
+    this.state = {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      rows: [
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        { id: v4(), text: 'Do something else', done: true }
+      ],
+      editItemId: null,
+      editItemText: ''
+    }
+  }
+
+  onChangeTextEditItem = editItemText => this.setState({ editItemText })
+
+  onPressHandler = id => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    const rows = this.state.rows.map(obj => {
+      if (obj.id === id) {
+        obj.done = !obj.done
+      }
+      return obj
+    })
+    this.setState({ rows })
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+      list.unshift({ id: v4(), text: e.nativeEvent.text, done: false })
+      this.setState({ rows: list, editItemText: '' })
+    } else {
+      const list = [...this.state.rows]
+      const editItem = list.find(obj => obj.id === this.state.editItemId)
+      const editItemIndex = list.indexOf(editItem)
+      const newItem = { ...editItem }
+      newItem.text = e.nativeEvent.text
+      list.splice(editItemIndex, 1, newItem)
+      this.setState({ rows: list, editItemId: null, editItemText: '' })
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    }
+  }
+
+  buttonEditClickListener = id => {
+    const editItem = this.state.rows.find(obj => obj.id === id)
+    this.setState({ editItemId: id, editItemText: editItem.text })
+  }
+
+  buttonDeleteClickListener = id => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+            list.splice(DeleteIndex, 1)
+            this.setState({ rows: list })
+          }
+        }
+      ],
+      { cancelable: false }
+    )
+  }
+
+  renderItem = ({ item }) => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+    return (
+      <View style={styles.view}>
+        <TextInputToDoList
+          onSubmitTextFromInput={this.onSubmitTextFromInput}
+          onChangeTextEditItem={this.onChangeTextEditItem}
+          editItemId={editItemId}
+          editItemText={editItemText}
+        />
+        <Text style={styles.counter}>
+          Done: {rows.filter(row => row.done).length} / {rows.length}
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: App.js [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+      const editItem = list.find(obj => obj.id === this.state.editItemId)
+      const editItemIndex = list.indexOf(editItem)
+      const newItem = { ...editItem }
+      newItem.text = e.nativeEvent.text
+      list.splice(editItemIndex, 1, newItem)
+      this.setState({ rows: list, editItemId: null, editItemText: '' })
+    }
+  }
+
+  buttonEditClickListener = id => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply App.js]: done

File: src/components/Todos.js [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+            onSubmitEditing={this.addNewItem} 
+            value={text}
+            onChangeText={text => this.setState({text})}           
+          />
+        </View>
+        <Text style={counter}>
+          Done: {this.doneItems(items)} / {this.countItems(items)}
+        </Text>
+        <ScrollView>
+          <TodoItem
```
Comment (zheoreh) [inline src/components/Todos.js]: @VeraPavlova18 `this.state.items.map(item => <TodoItem data={item} ...>)`

File: src/components/Todos.js [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+        items: prevState.items.map(item => ({
+          ...item,
+          done: item.id === id ? !item.done : item.done
+        }))
+      }
+    })
+  }
+
+  addNewItem = e => {
+    const text = e.nativeEvent.text
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply src/components/Todos.js]: +

File: src/components/Todos.js [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+      'Do you want delete it?',
+      [
+        {
+          text: 'Cancel',
+          onPress: () => {},
+          style: 'cancel'
+        },
+        {
+          text: 'OK',
+          onPress: () => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply src/components/Todos.js]: done

File: src/components/Todos.js (hunk 11 above)
Comment (VeraPavlova18) [reply src/components/Todos.js]: done

File: src/components/Todos.js [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+      return {
+        items: prevState.items.map(item => ({
+          ...item,
+          edit: item.id === id ? !item.edit : item.edit
+        }))
+      }
+    })
+  }
+
+  editItem = (id, e) => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply src/components/Todos.js]: done

File: src/components/Todos.js [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+        items: [
+          { id: v4(), text, done: false, edit: false },
+          ...prevState.items
+        ],
+        text: ''
+      }
+    })
+  }
+
+  onEdit = id => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply src/components/Todos.js]: done

File: src/components/TodoItem.js [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+import PropTypes from 'prop-types'
+
+export default class TodoItem extends Component {
+  state = {
+    edit: false,
+    text: ''
+  }
+
+  onEdit = () => {
+    this.setState(prevState => {
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply src/components/TodoItem.js]: done

File: src/components/TodoItem.js [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+    )
+
+    const textInput = (
+      <TextInput
+        style={styles.input}        
+        onSubmitEditing={() => {
+          editItem(data.id, text);
+          this.onEdit()
+        }} 
+        defaultValue={data.text}
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply src/components/TodoItem.js]: done

File: src/components/TodoItem.js [hunk 18]
```diff
@@ -0,0 +1,97 @@
+import React, {Component} from 'react'
+import { Text, View, Button, TextInput } from 'react-native'
+import PropTypes from 'prop-types'
+
+export default class TodoItem extends Component {
+  state = {
+    edit: false,
+    text: ''
```
[the remark this answers is not in the document]
Comment (VeraPavlova18) [reply src/components/TodoItem.js]: done

**Pull request merged**